nifi-nar-bundles/nifi-email-bundle/nifi-email-processors/src/main/java/org/apache/nifi/processors/email/AbstractEmailProcessor.java

```diff
@@ -0,0 +1,391 @@
+    public static final PropertyDescriptor USER = new PropertyDescriptor.Builder()
+            .name("User Name")
+            .description("User Name used for authentication and authorization with Email server.")
+            .required(true)
+            .expressionLanguageSupported(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+    public static final PropertyDescriptor PASSWORD = new PropertyDescriptor.Builder()
+            .name("Password")
+            .description("Password used for authentication and authorization with Email server.")
+            .required(true)
+            .expressionLanguageSupported(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .sensitive(true)
+            .build();
+    public static final PropertyDescriptor FOLDER = new PropertyDescriptor.Builder()
+            .name("Folder")
+            .description("Email folder to retrieve messages from (e.g., INBOX)")
+            .required(true)
+            .expressionLanguageSupported(true)
+            .defaultValue("INBOX")
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+    public static final PropertyDescriptor FETCH_SIZE = new PropertyDescriptor.Builder()
+            .name("Fetch Size")
+            .description("Specify the maximum number of Messages to fetch per call to Email Server.")
+            .required(true)
+            .expressionLanguageSupported(true)
+            .defaultValue("10")
+            .addValidator(StandardValidators.POSITIVE_INTEGER_VALIDATOR)
+            .build();
+    public static final PropertyDescriptor SHOULD_DELETE_MESSAGES = new PropertyDescriptor.Builder()
+            .name("Delete Messages")
+            .description("Specify whether mail messages should be deleted after retrieval.")
+            .required(true)
+            .allowableValues("true", "false")
+            .defaultValue("false")
+            .addValidator(StandardValidators.BOOLEAN_VALIDATOR)
+            .build();
+
+    static final Relationship REL_SUCCESS = new Relationship.Builder()
+            .name("success")
+            .description("All messages that are the are successfully received from Email server and converted to FlowFiles are routed to this relationship")
+            .build();
+
+    static List<PropertyDescriptor> SHARED_DESCRIPTORS = new ArrayList<>();
+
+    static Set<Relationship> SHARED_RELATIONSHIPS = new HashSet<>();
+
+    /*
+     * Will ensure that list of PropertyDescriptors is build only once, since
+     * all other lifecycle methods are invoked multiple times.
+     */
+    static {
+        SHARED_DESCRIPTORS.add(HOST);
+        SHARED_DESCRIPTORS.add(PORT);
+        SHARED_DESCRIPTORS.add(USER);
+        SHARED_DESCRIPTORS.add(PASSWORD);
+        SHARED_DESCRIPTORS.add(FOLDER);
+        SHARED_DESCRIPTORS.add(FETCH_SIZE);
+        SHARED_DESCRIPTORS.add(SHOULD_DELETE_MESSAGES);
```

Comment:
what do you think about the idea of adding a shared state + a property controlling the number of incorrect login attempts? This would prevent a processor with the incorrect password causing an account lockup against IMAP or POP3. 

That would mimic the behavior used by both Mozilla and Outlook will not blindly try to login once a password fails.